src/doc/src/reference/build-scripts.md

```diff
@@ -74,16 +74,25 @@ directory specified in the [`OUT_DIR` environment variable][build-env]. Scripts
 should not modify any files outside of that directory.
 
 Build scripts communicate with Cargo by printing to stdout. Cargo will
-interpret each line that starts with `cargo:` as an instruction that will
+interpret each line that starts with `cargo::` as an instruction that will
 influence compilation of the package. All other lines are ignored.
 
-> Note: The order of `cargo:` instructions printed by the build script *may*
+After Cargo v1.72.0, you can use the following syntaxes for setting Cargo keys:
+
+* `cargo::KEY=VALUE` for setting reserved keys.
+* `cargo::metadata=KEY=VALUE` for setting unreserved keys as metadata.
+
+> Note: The old invocation prefix `cargo:` (one colon only) is deprecated and planned to be removed.
+> To migrate, use two-colons prefix cargo:: instead.
+> If you were using cargo:KEY=VALUE for arbitrary links manifest key-value pair, use cargo::metadata=KEY=VALUE.
+>
```

Comment:
What is reserved keys? What do you mean by “as metadata”? There is no single word mentioning “reserved keys” in the current doc. And that we might not want users to spend time learning that kinda of an implementation detail. 

I think the whole paragraph can compile down into this (very opinioned tho).

```suggestion
> **Note:** The old invocation prefix `cargo:` (one colon only) is deprecated
> and won't get any new features. To migrate, use two-colons prefix `cargo::`,
> which was added in Rust 1.72. If you were using `cargo:KEY=VALUE` for
> arbitrary links manifest key-value pairs, it is encouraged to switch to
> `cargo::metadata=KEY=VALUE`. Stick to `cargo:` only if the support of Rust
> version older than 1.72 is required.

```

Mind that

* We usually just say `Rust 1.2.3` instead of `Cargo 1.2.3`, although they are pretty much the same at this time.
* It looks better when there is a newline between this paragraph and the next "The order of `cargo::`", since they discuss different topics.

<img width="300" alt="image" src="https://github.com/rust-lang/cargo/assets/14314532/52cc392a-945f-4c75-b26d-49abc8f54386">

Reply:
Thanks for your great suggestions! Updated!